rmw_zenoh_cpp/src/detail/rmw_context_impl_s.hpp

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DETAIL__RMW_CONTEXT_IMPL_S_HPP_
+#define DETAIL__RMW_CONTEXT_IMPL_S_HPP_
+
+#include <zenoh.h>
+
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <string>
+
+#include "graph_cache.hpp"
+#include "liveliness_utils.hpp"
+
+#include "rcutils/types.h"
+#include "rmw/rmw.h"
+
+///=============================================================================
+class rmw_context_impl_s final
+{
+public:
+  // Constructor that internally initializees the Zenoh session and other artifacts.
+  // Throws an std::runtime_error if any of the initializations fail.
+  // The construction will block until a Zenoh router is detected.
+  // TODO(Yadunund): Make this a non-blocking call by checking for the Zenoh
+  // router in a separate thread. Instead block when creating a node if router
+  // check has not succeeded.
+  rmw_context_impl_s(
+    const rcutils_allocator_t * allocator,
+    const std::size_t domain_id,
```

Comment:
include `<cstddef>`

Reply:
Since we're compiling with C++, `std::size_t` is part of the standard library and should not need any additional includes. Please correct me if i'm mistaken.